src/Disks/S3/DiskS3.cpp

```diff
@@ -942,20 +942,32 @@ void DiskS3::startup()
 
     LOG_INFO(&Poco::Logger::get("DiskS3"), "Starting up disk {}", name);
 
-    /// Find last revision.
+    if (readSchemaVersion(bucket, s3_root_path) < RESTORABLE_SCHEMA_VERSION)
+        migrateToRestorableSchema();
+
+    findLastRevision();
```

Comment:
Revision of migrated files will be 0, because its name not in format `r<rev><random_name>`, but only `<random_name>`. Is it correct?

Reply:
Yes, for old files revision is 0 by default.